purpleair-aqi.js

```diff
@@ -225,27 +303,42 @@ function getAQITrend({ v1: partLive, v3: partTime }) {
   return "arrow.left.and.right";
 }
 
+/**
+ * Constructs an SFSymbol from the given symbolName
+ *
+ * @param {string} symbolName
+ * @returns {object} SFSymbol
+ */
+function createSymbol(symbolName) {
+  const symbol = SFSymbol.named(symbolName);
+  symbol.applyFont(Font.systemFont(20));
+  return symbol;
+}
+
 async function run() {
   const listWidget = new ListWidget();
   listWidget.setPadding(10, 15, 10, 10);
 
   try {
-    console.log(`Using sensor ID: ${SENSOR_ID}`);
+    /** @type {LatLon} */
+    const deviceLocation = await Location.current();
```

Comment:
If someone passed in a location, you could avoid fetching the location if you move this method into `getSensorId()`

Reply:
This call can be expensive, so I'd definitely skip it if we can avoid it.

Reply:
I am definitely concerned about this, we should only look for a location if there's no sensor ID in the widget.